CMSgov/mint-app
# Easi 2266/follow a model

# EASI-2266

## Changes and Description

- Added mutations to add and remove favorite model plans
- `FavoriteCard` component and icon
- Pagination table for navigating on more than 3 favorites
- Favorite column on models table
- Toggling of favorites on table, card and readonly page icon
- Custom hook for grouping favorite mutations
- Unit/Cypress testing

<img width="1016" alt="Screen Shot 2022-09-15 at 10 58 02 AM" src="https://user-images.githubusercontent.com/95709965/190437609-b836cf4d-9970-44e6-99a7-e8fd82699516.png">

<img width="996" alt="Screen Shot 2022-09-15 at 10 54 37 AM" src="https://user-images.githubusercontent.com/95709965/190436793-9f5daefb-3801-40e6-a5fe-474547d1ca8e.png">

<img width="976" alt="Screen Shot 2022-09-15 at 10 54 57 AM" src="https://user-images.githubusercontent.com/95709965/190436877-1a65d89c-91fa-44b6-bddd-3243ef5e4277.png">

## How to test this change

- Seed models
- Navigate to /models
- Favorite any model in the table by clicking start icon
- Remove favorite from either the table or directly on the card
- Add more than 3 favorites and use pagination
- Navigate to readonly view /models/${modelID}/read-only/model-basics
- Toggle the favorite icon and verify change on /models

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
